src/realm/object-store/sync/sync_session.hpp

```diff
@@ -408,7 +408,7 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
 
     void download_fresh_realm(const sync::SessionErrorInfo& error_info)
         REQUIRES(!m_config_mutex, !m_state_mutex, !m_connection_state_mutex);
-    void handle_fresh_realm_downloaded(DBRef db, StatusWith<sync::SessionErrorInfo> error_info,
+    void handle_fresh_realm_downloaded(DBRef db, Status result, const sync::SessionErrorInfo& cr_error_info,
```

Comment:
Why did this change back?

Reply:
Because I needed access to both the `Status` that just occurred as well as the `SessionErrorInfo` for the originating client reset error. When it was using the `StatusWith`, you would only have access to one or the other...